File: Entregables/taller4/Main.java
import java.util.Scanner;

public class Main {

    public static boolean rEq(String a, String b) {
        if (a.equals(b)) {
            return true;
        }

        if (a.length() == 1 && b.length() == 1) {
            return a.equals(b);
        } else {
            if (a.length() % 2 == 0) {
                int m = a.length() / 2;
                String a1 = a.substring(0, m);
                String a2 = a.substring(m);
                String b1 = b.substring(0, m);
                String b2 = b.substring(m);
                return ((rEq(a1, b2) && rEq(a2, b1)) || (rEq(a1, b1) && rEq(a2, b2)));
            }
            return false;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String a = scanner.nextLine();
        String b = scanner.nextLine();

        boolean result = rEq(a, b);
        if (result) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }

        scanner.close();
    }
}

File: Entregables/taller4/main.cpp
#include<iostream>
#include<string>
using namespace std;

bool rEq(string a, string b) {
    if (a == b)
        return true;

    if (a.length() == 1 && b.length() == 1)
        return a == b;
    
    if (a.length() % 2 == 0) {
        int m = a.length() / 2;
        string a1 = a.substr(0, m);
        string a2 = a.substr(m);
        string b1 = b.substr(0, m);
        string b2 = b.substr(m);
        return ((rEq(a1, b2) && rEq(a2, b1)) || (rEq(a1, b1) && rEq(a2, b2)));
    }
    return false;
}

int main() {
    string a, b;
    cin >> a >> b;
    if (rEq(a, b))
        cout << "YES";
    else
        cout << "NO";
    return 0;
}
